### PR TITLE
LOOP-1356 Jhbate/pump simulator setup 

### PR DESCRIPTION
**Pull request by jh-bate**
- file names  camel case
- use of `ApplyOne`
- seperate pump simulator screen and tests

### CODE REVIEW

File: src/settingsScreen.js [hunk 1]
```diff
@@ -173,10 +176,8 @@ class SettingsScreen {
             await screen.Close();
         }
         if (values.CorrectionRanges) {
-            let screen = this.OpenCorrectionRangeScreen();
```
Comment (darinkrauss) [inline src/settingsScreen.js:176]: [nit] You don't have to open and close the screen first?

File: src/settingsScreen.js [hunk 2]
```diff
@@ -194,7 +195,6 @@ class SettingsScreen {
             let screen = this.OpenInsulinSensitivitiesScreen();
             await screen.ApplyAll(values.InsulinSensitivities);
             await screen.Save();
-            await screen.Close();
```
Comment (darinkrauss) [inline src/settingsScreen.js:197]: [nit] If `Save` also closes the screen, consider `SaveAndClose` to make it abundantly clear the open-and-close bookends.

File: src/subScreen/pumpSimulatorScreen.js [hunk 3]
```diff
@@ -0,0 +1,216 @@
...
+            await this.ErrorOnSuspendOff();
+        }
+        if (settings.batteryRemaining) {
+            await this.ApplyBatteryRemaining(settings.batteryRemaining);
+        }
+        if (settings.reservoirRemaining) {
+            await this.ApplyReservoirRemaining(settings.reservoirRemaining);
+        }
+    }
+    async ErrorOnBolusOn() {
```
Comment (darinkrauss) [inline src/subScreen/pumpSimulatorScreen.js]: [nit] Consider collapsing into one `SetErrorOnBolus` function with a parameter to indicate on (true), off (false), or nil (no change). Then the above `Apply` collapses down nicely.

File: src/subScreen/pumpSimulatorScreen.js [hunk 4]
```diff
@@ -0,0 +1,216 @@
...
+        if (await this._isErrorOnBolus()) {
+            return;
+        }
+        await this.ErrorOnBolusButton().tap();
+    }
+    async ErrorOnBolusOff() {
+        if (await this._isErrorOnBolus()) {
+            await this.ErrorOnBolusButton().tap();
+        }
+        return;
```
Comment (darinkrauss) [inline src/subScreen/pumpSimulatorScreen.js]: [nit] Unnecessary `return`.

File: tests/accessibility/settingsSuspendThresholdScreen.tests.js [hunk 5]
```diff
@@ -20,9 +20,9 @@ var settingsSuspendThresholdScreenAccessibilityTests = (test) => {
     it('has an info button', async () => {
         await expect(screen.InfoButton()).toExist();
     });
-    it('has a cancel button', async () => {
-        await expect(screen.CancelButton()).toExist();
-    });
+    // it('has a cancel button', async () => {
```
Comment (darinkrauss) [inline tests/accessibility/settingsSuspendThresholdScreen.tests.js:23]: [nit] TODO so you don't forget to uncomment at a later point?

**Review by darinkrauss — APPROVED** [review]
LGTM! 🏈

File: src/settingsScreen.js (hunk 2 above)
Comment (jh-bate) [reply src/settingsScreen.js:197]: good point

File: src/subScreen/pumpSimulatorScreen.js (hunk 3 above)
Comment (jh-bate) [reply src/subScreen/pumpSimulatorScreen.js]: yeah good idea

**Pull request merged**